driver/controller.go

```diff
@@ -806,11 +807,42 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	return listResp, nil
 }
 
+// ControllerExpandVolume is called from the resizer to increase the volume size.
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	d.log.WithField("method", "controller_expand_volume").
-		Info("controller expand volume called")
+	ll := d.log.WithField("method", "controller_expand_volume")
+	ll.Info("controller expand volume called")
+	volID := req.GetVolumeId()
 
-	return nil, status.Error(codes.Unimplemented, "")
+	if len(volID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID missing in request")
+	}
+	volume, _, err := d.storage.GetVolume(ctx, volID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume could not retrieve existing volume: %v", err)
+	}
+	ll.Info("extracting storage from requested capacity range")
+	resizeBytes, err := extractStorage(req.GetCapacityRange())
+	if err != nil {
+		return nil, status.Errorf(codes.OutOfRange, "ControllerExpandVolume invalid capacity range: %v", err)
+	}
+	resizeGigaBytes := resizeBytes / giB
+
+	if resizeGigaBytes <= volume.SizeGigaBytes {
+		return &csi.ControllerExpandVolumeResponse{CapacityBytes: volume.SizeGigaBytes * giB, NodeExpansionRequired: true}, nil
```

Comment:
Consider this scenario: 
- user creates a pvc initially with 5Gi capacity. 
- A dynamic PV is created with 5Gi capacity
- Using the control panel, the volume's capacity is increased to 20Gi
- The PVC is then edited to increase the capacity to 10Gi
- NodeExpandVolume should be called to resize the FS to (10Gi | 20Gi), right? 

Increasing the volume size in the control panel does not mean that the PVC capacity will increase. When the user requests that the PVC capacity be increased, we should make a call to `NodeExpandVolume`. We discussed that this shouldn't cause an error while requesting a resize. 

We were discussing whether the `CapacityBytes` we return here should be the what the user requested or volume capacity. I have provided the volume capacity here so that the PV and PVC capacities become consistent when the resize is requested. Not sure if it is alright for the two capacities to be inconsistent for dynamically provisioned PVs.

Reply:
Thanks for explaining, that does make sense. Perhaps we should test this scenario manually to make sure it really works as expected and doesn't break things. WDYT?

(And if it works as expected, we should add a quick comment above the line in question to describe the reasoning to future code readers.)